standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/TransactionalValidationListener.java

```diff
@@ -146,6 +146,12 @@ private void handleAlterTableTransactionalProp(PreAlterTableEvent context) throw
     }
 
     if (transactionalValuePresent) {
+      if (oldTable.getTableType().equals(TableType.MANAGED_TABLE.toString())
+              && newTable.getTableType().equals(TableType.EXTERNAL_TABLE.toString())) {
```

Comment:
Shouldn't we also check oldTransactionValuePresent? We shouldn't allow an old transactional managed table to be converted to non-transactional external table.

Reply:
Good point, will incorporate it.

Reply:
In fact, this code handles both conversion. If transactional key(not value) exist, it reaches this flow. So, it handles conversion from old transactional managed table to be converted to non-transactional external table.